src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -276,10 +276,21 @@ public override async Task RunAsync()
                 }
             }
 
-            // Check if azure-functions-worker exists in requirements.txt for Python function app
+            // Special checks for python dependencies
             if (workerRuntime == WorkerRuntime.python)
             {
+                // Check if azure-functions-worker exists in requirements.txt for Python function app
                 await PythonHelpers.WarnIfAzureFunctionsWorkerInRequirementsTxt();
+                // Check if remote LinuxFxVersion exists and is different from local version
+                if (!string.IsNullOrEmpty(functionApp.LinuxFxVersion))
```

Comment:
Also my need an `else` statement, since when `functionApp.LinuxFxVersion` is empty, that stands for `Python|3.6`.